cvat-core/src/server-proxy.js

```diff
@@ -612,41 +612,64 @@
-                                proxy: config.proxy,
-                            });
-                            if (response.status === 202) {
-                                if (onUpdate && response.data.message !== '') {
-                                    onUpdate(response.data.message, response.data.progress || 0);
+                try {
+                    await Axios.post(url,
+                        new FormData(), {
+                            params,
+                            proxy: config.proxy,
+                            headers: { 'Upload-Start': true },
+                        });
+                    await chunkUpload(file, uploadConfig);
+                    await Axios.post(url,
+                        new FormData(), {
+                            params,
+                            proxy: config.proxy,
+                            headers: { 'Upload-Finish': true },
+                        });
+                    return new Promise((resolve, reject) => {
+                        async function requestStatus() {
+                            try {
+                                const response = await Axios.get(`${url}?action=import_status`, {
+                                    params,
+                                    proxy: config.proxy,
+                                });
+                                if (response.status === 202) {
+                                    if (onUpdate && response.data.message !== '') {
+                                        onUpdate(response.data.message, response.data.progress || 0);
+                                    }
+                                    setTimeout(requestStatus, 3000);
+                                } else if (response.status === 201) {
+                                    resolve();
+                                } else {
+                                    reject(generateError(response));
                                 }
-                                setTimeout(requestStatus, 3000);
-                            } else if (response.status === 201) {
-                                resolve();
-                            } else {
-                                reject(generateError(response));
+                            } catch (error) {
+                                reject(generateError(error));
                             }
-                        } catch (error) {
-                            reject(generateError(error));
                         }
-                    }
-
-                    Axios.post(`${url}?format=${format}`, formData, {
-                        proxy: config.proxy,
-                    }).then(() => {
                         setTimeout(requestStatus, 2000);
-                    }).catch((error) => {
-                        reject(generateError(error));
                     });
-                });
+                } catch (errorData) {
+                    generateError(errorData);
```

Comment:
Probably generated error should be raised
And we should not return null in this case.

Reply:
Will such lines be better?
```
  generateError(errorData);
  throw errorData;
```
If we dont throw or return something, eslint `consistent-return` error occurs

Reply:
Yep, looks like we do the same in other server methods.